bigquery_etl/view/__init__.py

```diff
@@ -291,6 +280,30 @@ def has_changes(self, target_project=None):
                 print(f"view {target_view_id} will change: labels do not match")
                 return True
 
+        schema_file = Path(self.path).parent / "schema.yaml"
+        table_schema = Schema.from_bigquery_schema(table.schema)
+        # check schema based on schema file
+        if schema_file.is_file():
+            view_schema = Schema.from_schema_file(schema_file)
+        else:  # check schema based on dry run results
+            try:
+                query_job = client.query(
+                    query=self.content,
+                    job_config=bigquery.QueryJobConfig(
+                        dry_run=True, use_legacy_sql=False
+                    ),
+                )
+                view_schema = Schema.from_bigquery_schema(query_job.schema)
+            except Forbidden:
+                print(
+                    f"Missing permission to dry run view {target_view_id} to get schema"
+                )
+                view_schema = None
```

Comment:
I guess the downside here is that we no longer update views that have any access restrictions. Maybe we should always update them instead?